lib/manager/components/DeploymentsPanel.js

```diff
@@ -137,17 +143,32 @@ class DeploymentsList extends Component<ListProps, State> {
     const {deployments} = project
     const {searchText} = this.state
     const {pinnedDeploymentId} = project
-    const pinnedDeployment = deployments &&
-      pinnedDeploymentId &&
-      deployments.find(deployment => deployment.id === pinnedDeploymentId)
     const visibleDeployments: Array<Deployment> = deployments
-      ? deployments.filter(
-        deployment =>
+      ? deployments
+        // Filter deployments by search text.
+        .filter(deployment =>
           deployment.name
             .toLowerCase()
-            .indexOf((searchText || '').toLowerCase()) !== -1 &&
-          deployment.id !== pinnedDeploymentId
-      )
+            .indexOf((searchText || '').toLowerCase()) !== -1
+        )
+        .sort((a, b) => {
+          // If creating deployment, pin to top.
+          if (!a.name) return -1
+          if (!b.name) return 1
+          // Ensure pinned deployment is first element in list.
+          if (b.id === pinnedDeploymentId) return 1
+          if (a.id === pinnedDeploymentId) return -1
+          // Otherwise, sort by most recent last deployed date and then most recent
+          // created date (ensuring deployments never deployed show up at end of
+          // list).
+          // TODO: Refactor to allow for sorting on multiple fields? This may be
+          //  overkill for deployments.
```

Comment:
I agree this may be a bit much to have multi-level sorting. Perhpas it would be possible to implement a sorting button like was done for the project feeds table.

Reply:
Agreed, I'll create an issue from this comment.